Yosemite/Yosemite/Model/Storage/Coupon+ReadOnlyConvertible.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+import Storage
+
+
+// MARK: - Storage.Coupon: ReadOnlyConvertible
+//
+extension Storage.Coupon: ReadOnlyConvertible {
+
+    /// Updates the `Storage.Coupon` from the ReadOnly representation (`Networking.Coupon`)
+    ///
+    public func update(with coupon: Yosemite.Coupon) {
+        siteID = coupon.siteID
+        couponID = coupon.couponID
+        code = coupon.code
+        amount = coupon.amount
+        dateCreated = coupon.dateCreated
+        dateModified = coupon.dateModified
+        discountType = coupon.discountType.rawValue
+        fullDescription = coupon.description
+        dateExpires = coupon.dateExpires
+        usageCount = coupon.usageCount
+        individualUse = coupon.individualUse
+        products = coupon.productIds
+        excludedProducts = coupon.excludedProductIds
+        usageLimit = Int64(coupon.usageLimit ?? 0)
+        usageLimitPerUser = Int64(coupon.usageLimitPerUser ?? 0)
+        limitUsageToXItems = Int64(coupon.limitUsageToXItems ?? 0)
+        freeShipping = coupon.freeShipping
+        productCategories = coupon.productCategories
+        excludedProductCategories = coupon.excludedProductCategories
+        excludeSaleItems = coupon.excludeSaleItems
+        minimumAmount = coupon.minimumAmount
+        maximumAmount = coupon.maximumAmount
+        emailRestrictions = coupon.emailRestrictions
+        usedBy = coupon.usedBy
+    }
+
+    /// Returns a ReadOnly (`Networking.Coupon`) version of the `Storage.Coupon`
+    ///
+    public func toReadOnly() -> Yosemite.Coupon {
+        return Yosemite.Coupon(siteID: siteID,
```

Comment:
Minor nit
```suggestion
        return .init(siteID: siteID,
```

Reply:
Using `.init` gives the error `The compiler is unable to type-check this expression in reasonable time; try breaking up the expression into distinct sub-expressions`

However it doesn't need the context of `Yosemite`, so I've removed that to shorten it a bit.